FindYourMovie/Controllers/MovieViewController.swift

```diff
@@ -0,0 +1,222 @@
+        
+        let alert = UIAlertController(title: "Error on Search", message: errorToDisplay.getCustomError(), preferredStyle: .alert)
+        let ok = UIAlertAction(title: "OK", style: .default)
+        alert.addAction(ok)
+        present(alert, animated: true, completion: nil)
+        
+    }
+    
+    /// Updates the last searches done by the user. No duplicates are added and the size is kept to 10
+    ///
+    /// - Parameter movieName: Name of the movie to add to the list.
+    private func updateLastSearchResults(movieName: String) {
+        if !last10searches.contains(movieName){
+            if(self.last10searches.count == 10){
+                self.last10searches.removeLast()
+            }
+            self.last10searches.insert(movieName.replacingOccurrences(of: "+", with: " "), at: 0)
+            self.defaults.set(self.last10searches, forKey: "lastSearches")
+        }
+        
+        
+    }
+    
+}
+
+// MARK: - Search Bar Delegate
+//
+extension MovieViewController: UISearchBarDelegate {
+    
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        
+        searchBar.resignFirstResponder()
+        guard let searchText = searchBar.text, !searchText.isEmpty else {
+            return
+        }
+        requestMovies(movieName: searchText)
+        searchBar.text = ""
+        searchController.dismiss(animated: true, completion: nil)
+        
+    }
+    
+    
+}
+
+// MARK: - UISearchControllerDelegate
+
+extension MovieViewController: UISearchControllerDelegate {
+    
+    func presentSearchController(_ searchController: UISearchController) {
+        if #available(iOS 13.0, *) {
+            searchController.showsSearchResultsController = true
+        } else {
+            show(searchController.searchResultsController!, sender: self )
+        }
+        
+    }
+}
+
+// MARK: - UISearchResultsUpdating
+
+extension MovieViewController: UISearchResultsUpdating {
+    
+    // Called when the search bar's text has changed or when the search bar becomes first responder.
+    func updateSearchResults(for searchController: UISearchController) {
+        // Update the resultsController's filtered items based on the search terms and suggested search token.
+        self.last10searches = defaults.array(forKey: "lastSearches") as! [String]
+        let lastSearches = self.last10searches
+        
+        // Apply the filtered results to the search results table.
+        if let resultsController = searchController.searchResultsController as? MovieSearchController {
+            resultsController.parentController = self
+            resultsController.searches = lastSearches
+            resultsController.tableView.reloadData()
+        }
+    }
+    
+}
+
+
+extension UIImageView {
+    func load(string: String?) {
+        DispatchQueue.global().async { [weak self] in
+            guard let url = string else {
+                if #available(iOS 13.0, *) {
```

Comment:
This is kind of a large else condition for a guard -- would be easier to follow as

```
guard let url = string else {
   doTheThing()
   return
}
```